media-gfx/freecad/freecad-0.16.ebuild

```diff
@@ -0,0 +1,143 @@
+# https://www.gentoo.org/foundation/en/minutes/2011/20110220_trustees.meeting_log.txt
+RESTRICT="mirror"
+
+# TODO:
+#   DEPEND and RDEPEND:
+#		salome-smesh - science overlay
+#		zipio++ - not in portage yet
+
+S="${WORKDIR}/FreeCAD-${PV}"
+
+DOCS=(README.md ChangeLog.txt)
+
+pkg_setup() {
+	fortran-2_pkg_setup
+	python-single-r1_pkg_setup
+
+	[ -z "${CASROOT}" ] && die "empty \$CASROOT, run eselect opencascade set or define otherwise"
+}
+
+src_prepare() {
+	epatch "${FILESDIR}"/${PN}-0.14.3702-install-paths.patch
+
+	#bug 518996
+	sed -e "/LibDir = /s:'lib':'"$(get_libdir)"':g" \
+		-i src/App/FreeCADInit.py || die
+	eapply_user
+
+}
+
+src_configure() {
+	export QT_SELECT=4
+	local mycmakeargs=(
+		-DOCC_INCLUDE_DIR="${CASROOT}"/inc
+		-DOCC_INCLUDE_PATH="${CASROOT}"/inc
+		-DOCC_LIBRARY="${CASROOT}"/lib/libTKernel.so
+		-DOCC_LIBRARY_DIR="${CASROOT}"/lib
+		-DOCC_LIB_PATH="${CASROOT}"/lib
+		-DCOIN3D_INCLUDE_DIR="${EROOT}"usr/include/coin
+		-DCOIN3D_LIBRARY="${EROOT}"usr/$(get_libdir)/libCoin.so
+		-DSOQT_LIBRARY="${EROOT}"usr/$(get_libdir)/libSoQt.so
+		-DSOQT_INCLUDE_PATH="${EROOT}"usr/include/coin
+		-DCMAKE_INSTALL_PREFIX="${EROOT}"usr/$(get_libdir)/${P}
+		-DCMAKE_INSTALL_DATADIR="${EROOT}"usr/share/${P}/
+		-DCMAKE_INSTALL_DOCDIR="${EROOT}"usr/share/doc/${P}/
+		-DCMAKE_INSTALL_INCLUDEDIR="${EROOT}"usr/include/${P}/
+	)
+
+	# TODO to remove embedded dependencies:
+	#
+	#	-DFREECAD_USE_EXTERNAL_ZIPIOS="ON" -- this option needs zipios++ but it's not yet in portage so the embedded zipios++
+	#                (under src/zipios++) will be used
+	#	salomesmesh is in 3rdparty but upstream's find_package function is not complete yet to compile against external version
+	#                (external salomesmesh is available in "science" overlay)
+
+	cmake-utils_src_configure
+	ewarn "${P} will be built against opencascade version ${CASROOT}"
+}
+
+src_install() {
+	cmake-utils_src_install
+
+	prune_libtool_files
+
+	make_wrapper FreeCAD \
+		"${EROOT}"usr/$(get_libdir)/${P}/bin/FreeCAD \
+		"" "${EROOT}"usr/$(get_libdir)/${P}/lib
+	make_wrapper FreeCADCmd \
+		"${EROOT}"usr/$(get_libdir)/${P}/bin/FreeCADCmd \
+		"" "${EROOT}"usr/$(get_libdir)/${P}/lib
+
+	make_desktop_entry FreeCAD "FreeCAD" "" "" "MimeType=application/x-extension-fcstd;"
+
+	# install mimetype for FreeCAD files
+	insinto /usr/share/mime/packages
+	newins "${FILESDIR}"/${PN}.sharedmimeinfo "${PN}.xml"
+
+	# install icons to correct place rather than /usr/share/freecad
+	pushd "${ED}/usr/share/${P}"
+	for size in 16 32 48 64; do
+		newicon -s ${size} freecad-icon-${size}.png freecad.png
+	done
+	doicon -s scalable freecad.svg
+	newicon -s 64 -c mimetypes freecad-doc.png application-x-extension-fcstd.png
+	popd
+
+	# disable compression of QT assistant help files
+	>> "${ED}"usr/share/doc/${P}/freecad.qhc.ecompress.skip
+	>> "${ED}"usr/share/doc/${P}/freecad.qch.ecompress.skip
+
+	python_optimize "${ED}"usr/{$(get_libdir),share}/${P}/Mod/
```

Comment:
Please use `"${ED%/}"/` style everywhere, as it is more forward compatible (last section, https://devmanual.gentoo.org/ebuild-writing/variables/)